src/test/kotlin/com/ampnet/web3provider/controller/ProviderControllerTest.kt

```diff
@@ -108,57 +105,53 @@ class ProviderControllerTest : ControllerTestBase() {
-            Mockito.`when`(defaultProviderService.getResponse(testContext.jsonRpcRequest))
-                .thenReturn(ProviderResponse(testContext.jsonRpcRequest, hexStringResult))
+            testContext.jsonRpcRequestJson = objectMapper.writeValueAsString(testContext.jsonRpcRequest)
+            mockDefaultProviderResponse(
+                testContext.jsonRpcRequestJson, generateProviderResponse(testContext.jsonRpcRequest, hexValue)
+            )
         }
 
         verify("Provider service returns correct result") {
-            val json =
-                """
-                    {
-                        "id":"1",
-                        "jsonrpc":"2.0",
-                        "method": "eth_chainId",
-                        "params":[]
-                    }
-                """.trimIndent()
             val result = mockMvc.perform(
-                post("/api").content(json).contentType(MediaType.APPLICATION_JSON)
+                post("/api").content(testContext.jsonRpcRequestJson).contentType(MediaType.APPLICATION_JSON)
             ).andExpect(status().isOk).andReturn()
             val response: ProviderResponse = objectMapper.readValue(result.response.contentAsString)
-            assertThat(response.result).isEqualTo(hexStringResult)
+            assertThat(response.result).isEqualTo(hexValue)
             assertThat(response.jsonrpc).isEqualTo(testContext.jsonRpcRequest.jsonrpc)
             assertThat(response.id).isEqualTo(testContext.jsonRpcRequest.id)
         }
         verify("Chain id is saved inside the cache") {
             val chainId = redisRepository.getCache(RedisEntity.CHAIN_ID.methodName, RedisEntity.CHAIN_ID.methodName)
-            assertThat(chainId).isEqualTo(hexStringResult)
+            assertThat(chainId).isEqualTo(hexValue)
         }
+        verify("Rest template called mocked server") { mockServer.verify() }
     }
 
     @Test
     fun mustBeAbleToGetTransactionByHash() {
         suppose("Default provider service will return transaction result") {
+            testContext.transaction = createTransaction()
             testContext.jsonRpcRequest = JsonRpcRequest(
                 "2.0", RedisEntity.TRANSACTION_BY_HASH.methodName, listOf(txHash), "1"
             )
-            testContext.transaction = createTransaction()
-            Mockito.`when`(defaultProviderService.getResponse(testContext.jsonRpcRequest))
-                .thenReturn(ProviderResponse(testContext.jsonRpcRequest, testContext.transaction))
+            testContext.jsonRpcRequestJson = objectMapper.writeValueAsString(testContext.jsonRpcRequest)
+            val response = """
+            {
+                "id":"${testContext.jsonRpcRequest.id}",
+                "jsonrpc":"${testContext.jsonRpcRequest.jsonrpc}",
+                "method": "${testContext.jsonRpcRequest.method}",
+                "result": ${objectMapper.writeValueAsString(testContext.transaction)}
+            }
+            """.trimIndent()
```

Comment:
Why not use function `generateProviderResponse` to generate this response?

Reply:
In case result is just a string ```"result": "$result"``` works with the "" around $result. In this test case it is a json object and it doesn't work with the "" so thats why I didn't use the function to generate the response.